src/python/WMCore/JobSplitting/EventBased.py

```diff
@@ -89,7 +91,7 @@ def algorithm(self, *args, **kwargs):
                                         self.subscription.workflowName(), self.subscription['id'])
                     for f in fileList:
                         lumiDict = fileLumis.get(f['id'], {})
-                        for run in lumiDict.keys():
+                        for run in list(lumiDict.keys()):
```

Comment:
```
for run in list(lumiDict):
```
?

Reply:
Uh, you are right! also `for run in lumiDict` should be ok. Let me check if I left this on purpose (strange, i should have added a comment) or if I just missed this change.
Thanks again!

Reply:
Ok, no unit test broke, so I really missed this one too! thanks again! :)